default-02      (100s ago)  changes to: Add darn() method to SecureRandom

diff --git a/src/hotspot/cpu/ppc/templateInterpreterGenerator_ppc.cpp b/src/hotspot/cpu/ppc/templateInterpreterGenerator_ppc.cpp
--- a/src/hotspot/cpu/ppc/templateInterpreterGenerator_ppc.cpp
+++ b/src/hotspot/cpu/ppc/templateInterpreterGenerator_ppc.cpp
@@ -1833,6 +1833,16 @@
   return start;
 }
 
+address TemplateInterpreterGenerator::generate_securerandom_darn_entry() {
+  address _pc = __ pc();
+  // It will generate a SIGILL on POWER8 machines...
+  __ darn(R3_RET);
+  __ blr();
+
+  return _pc;
+}
+
+
 // CRC32 Intrinsics.
 //
 // Contract on scratch and work registers.
diff --git a/src/hotspot/share/classfile/vmSymbols.hpp b/src/hotspot/share/classfile/vmSymbols.hpp
--- a/src/hotspot/share/classfile/vmSymbols.hpp
+++ b/src/hotspot/share/classfile/vmSymbols.hpp
@@ -776,6 +776,11 @@
   do_intrinsic(_full_name, gromero, full_name_name, void_method_signature, F_S)   \
   do_name(full_name_name, "full_name")                                            \
                                                                                   \
+  /* darn */                                                                      \
+  do_class(securerandom, "java/security/SecureRandom")                                        \
+  do_intrinsic(_darn, securerandom, darn_name, serializePropertiesToByteArray_signature, F_S) \
+  do_name(darn_name, "darn")                                                                  \
+                                                                                              \
   /* here are the math names, all together: */                                                                          \
   do_name(abs_name,"abs")       do_name(sin_name,"sin")         do_name(cos_name,"cos")                                 \
   do_name(tan_name,"tan")       do_name(atan2_name,"atan2")     do_name(sqrt_name,"sqrt")                               \
diff --git a/src/hotspot/share/interpreter/abstractInterpreter.cpp b/src/hotspot/share/interpreter/abstractInterpreter.cpp
--- a/src/hotspot/share/interpreter/abstractInterpreter.cpp
+++ b/src/hotspot/share/interpreter/abstractInterpreter.cpp
@@ -138,6 +138,8 @@
     case vmIntrinsics::_doubleToRawLongBits: return java_lang_Double_doubleToRawLongBits;
     // gromero
     case vmIntrinsics::_full_name: return gromero_full_name;
+    // darn
+    case vmIntrinsics::_darn:      return securerandom_darn;
     default:                                 break;
   }
 #endif // CC_INTERP
diff --git a/src/hotspot/share/interpreter/abstractInterpreter.hpp b/src/hotspot/share/interpreter/abstractInterpreter.hpp
--- a/src/hotspot/share/interpreter/abstractInterpreter.hpp
+++ b/src/hotspot/share/interpreter/abstractInterpreter.hpp
@@ -89,6 +89,7 @@
     java_lang_Double_longBitsToDouble,                          // implementation of java.lang.Double.longBitsToDouble()
     java_lang_Double_doubleToRawLongBits,                       // implementation of java.lang.Double.doubleToRawLongBits()
     gromero_full_name,                                          // implementation of gromero.full_name() method ;-)
+    securerandom_darn,                                          // implementation of darn method...
     number_of_method_entries,
     invalid = -1
   };
diff --git a/src/hotspot/share/interpreter/templateInterpreterGenerator.cpp b/src/hotspot/share/interpreter/templateInterpreterGenerator.cpp
--- a/src/hotspot/share/interpreter/templateInterpreterGenerator.cpp
+++ b/src/hotspot/share/interpreter/templateInterpreterGenerator.cpp
@@ -227,6 +227,7 @@
   method_entry(java_lang_Double_doubleToRawLongBits);
 
   method_entry(gromero_full_name)
+  method_entry(securerandom_darn)
 
 #undef method_entry
 
@@ -412,6 +413,7 @@
 
   switch (kind) {
   case Interpreter::gromero_full_name      : entry_point = generate_full_name_entry();            break;
+  case Interpreter::securerandom_darn      : entry_point = generate_securerandom_darn_entry();    break;
   case Interpreter::zerolocals             :                                          break;
   case Interpreter::zerolocals_synchronized:                synchronized = true;      break;
   case Interpreter::native                 : native = true;                           break;
diff --git a/src/hotspot/share/interpreter/templateInterpreterGenerator.hpp b/src/hotspot/share/interpreter/templateInterpreterGenerator.hpp
--- a/src/hotspot/share/interpreter/templateInterpreterGenerator.hpp
+++ b/src/hotspot/share/interpreter/templateInterpreterGenerator.hpp
@@ -88,6 +88,8 @@
 
   // gromero
   address generate_full_name_entry(void);
+  // darn
+  address generate_securerandom_darn_entry(void);
 
   address generate_normal_entry(bool synchronized);
   address generate_native_entry(bool synchronized);
